warehouse/manage/views/__init__.py

```diff
@@ -1435,6 +1436,117 @@ def manage_project_oidc_publishers_prefill(self):
 
         return self.manage_project_oidc_publishers()
 
+    @view_config(
+        request_method="GET", request_param=ConstrainEnvirormentForm.__params__
+    )
+    def manage_project_oidc_publisher_constrain_environment(self):
```

Comment:
I don't think we should have a view that can be triggered from a GET request that modifies the publisher (or anything, really). Even though this is a relatively minor change, it could be used semi-maliciously but also breaks a general pattern we have of links being non-mutating.

Ideally this would go from email link -> `GET` interstitial that confirms the changes -> `POST` that applies the changes (the last two can be the same view, varied by the request method).

Reply:
done! I added a modal for confirming the changes, which then applies them POST

<img width="400" alt="image" src="https://github.com/user-attachments/assets/71dfb581-67ba-4165-8693-05b8a7befd4e" />

(see the updated PR description, and the last commit)